### PR TITLE
Fix on Mojave, there's no Keycastr in the Security & Privacy Panel

### PR DESCRIPTION
**Pull request by NicholasTD07**
This fixes the issue #129.

### CODE REVIEW

**Comment by akitchen** [comment]
Hello - what does this fix? I already get the error message as expected on the latest macOS

**Comment by akitchen** [comment]
I see now, this API brings up a native dialog and adds KeyCastr to the accessibility apps list (but the user still needs to check the box)

we can't take this change as-is since it breaks the current lifecycle pattern when starting the app for the first time, but it would be good to refactor to use this instead of the (broken) Applescript hack we currently have.

**Comment by NicholasTD07** [comment]
@akitchen Yes. Sorry. I should have mentioned this or included a screenshot.

File: keycastr/KCKeyboardTap.m [hunk 1]
```diff
@@ -87,7 +87,20 @@ -(BOOL) installTapWithError:(NSError **)error {
     if (tapInstalled) {
         return YES;
     }
-    
+
+    // 10.9 and after
+    if (AXIsProcessTrustedWithOptions != NULL) {
```
Comment (akitchen) [inline keycastr/KCKeyboardTap.m]: 10.9 is our minimum supported version in the project file, so this check probably isn't needed. Also, this style generates a warning

File: keycastr/KCKeyboardTap.m [hunk 2]
```diff
@@ -87,7 +87,20 @@ -(BOOL) installTapWithError:(NSError **)error {
...
+        NSDictionary *options = @{(__bridge id) kAXTrustedCheckOptionPrompt : @YES};
+        BOOL accessibilityEnabled = AXIsProcessTrustedWithOptions((__bridge CFDictionaryRef) options);
+
+        if (!accessibilityEnabled) {
+            if (error != NULL) {
+                *error = [self constructErrorWithDescription:@"Could not create keyDown event tap!"];
```
Comment (akitchen) [inline keycastr/KCKeyboardTap.m]: We'll have to revisit the error messaging for this check failing. Also, this error interferes with the native dialog generated during the call to `AXIsProcessTrustedWithOptions()`

File: keycastr/KCKeyboardTap.m [hunk 3]
```diff
@@ -87,7 +87,20 @@ -(BOOL) installTapWithError:(NSError **)error {
...
+            }
+            return NO;
```
Comment (akitchen) [inline keycastr/KCKeyboardTap.m]: This results in the app exiting, which tears down the native dialog generated by `AXIsProcessTrustedWithOptions()`. In order to use this API (which we should do!) we need to refactor to wait to exit the app until the user interacts with that dialog, if possible.

**Comment by akitchen** [comment]
Thank you for this, @NicholasTD07 . I added a few comments based on the brief time I spent testing this code. 

Do you want to try to refactor this to improve the first time startup experience? I believe this codepath may also be triggered when the KeyCastr settings change, although I haven't tested it exhaustively.

**Comment by NicholasTD07** [comment]
Sorry... I created this PR and then forgot about it... Do we still need this PR? I mean has this change made into the codebase through some other PRs already?

**Comment by NicholasTD07** [comment]
Closing this PR for now unless it is needed. Feel free to re-open it.

**Pull request closed without merging**